tests/loader/test_hilti_loader.py

```diff
@@ -91,28 +76,39 @@ def test_get_absolute_pose_priors(self) -> None:
         #     2,
         # )
 
-    def test_get_valid_pairs(self) -> None:
-        i1: int = 5  # cam 0 rig 1
-
-        # 0, 1, 3 from other rigs
-        # 1, 3 from the same rig
-        expected_i2s = {
-            0,  # cam 0 rig 0
-            1,  # cam 1 rig 0
-            3,  # cam 3 rig 0
-            6,  # cam 1 rig 1
-            8,  # cam 3 rig 1
-            10,  # cam 0 rig 2
-            11,  # cam 1 rig 2
-            13,  # cam 3 rig 2
-        }
-
-        for i2 in range(len(self.loader)):
-            print(i1, i2)
-            if i2 in expected_i2s:
-                if i1 < i2:
-                    self.assertTrue(self.loader.is_valid_pair(i1, i2))
-                else:
-                    self.assertTrue(self.loader.is_valid_pair(i2, i1))
-            else:
-                self.assertFalse(self.loader.is_valid_pair(i1, i2))
+    def test_number_of_relative_pose_priors(self) -> None:
+        """Check that 3 relative constraints translate into many relative pose priors."""
+        # Just give 3 pairs
+        pairs = [
+            (0, 1),
+            (0, 3),
+            (0, 5),
+        ]
+        expected = [
+            (0, 1),
+            (0, 3),
+            (0, 5),
+            (2, 0),
+            (2, 1),
+            (2, 3),
+            (2, 4),
+            (7, 5),
+            (7, 6),
+            (7, 8),
+            (7, 9),
+            (12, 10),
+            (12, 11),
+            (12, 13),
+            (12, 14),
```

Comment:
Can we stick with the scheme that (i1, i2) in the keys always have i1 < i2. I am not sure if this is needed, but I am worried something might break silently and cause performance degradation.

Reply:
Yes, thanks for noticing this, will fix!